packages/sdk/contractkit/src/kit.test.ts

```diff
@@ -100,3 +102,24 @@ describe('kit.sendTransactionObject()', () => {
     })
   })
 })
+
+describe('newKitWithApiKey()', () => {
+  const kit = newKitWithApiKey('http://', 'key')
+  const mockSetRequestHeader = jest.fn()
+  XMLHttpRequest.prototype.setRequestHeader = mockSetRequestHeader
```

Comment:
this is cool!

Reply:
haha I spent 99% of the time on this code change trying to figure out this silly mock